src/components/layouts/FilterBar.tsx

```diff
@@ -4,29 +4,77 @@ import { Avatar, Box, IconButton, Stack, Typography } from '@mui/material'
 import { AppMargin, SizeofAppMargin } from '@/hoc/AppMargin'
 import { useState } from 'react'
 import store from '@/redux/store'
-import { setFilteredAsignee, setFilteredTasks, setViewSettings } from '@/redux/features/taskBoardSlice'
+import { setViewSettings } from '@/redux/features/taskBoardSlice'
 import SearchBar from '@/components/searchBar'
 import Selector, { SelectorType } from '@/components/inputs/Selector'
 import { useHandleSelectorComponent } from '@/hooks/useHandleSelectorComponent'
 import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
 import { useSelector } from 'react-redux'
-import { IAssigneeCombined, View } from '@/types/interfaces'
-import { CrossIcon, FilterByAsigneeIcon } from '@/icons'
+import { FilterOptions, FilterOptionsKeywords, IAssigneeCombined, IFilterOptions, View } from '@/types/interfaces'
+import { FilterByAsigneeIcon } from '@/icons'
 import { ViewModeSelector } from '../inputs/ViewModeSelector'
 import { FilterByAssigneeBtn } from '../buttons/FilterByAssigneeBtn'
+import FilterButtonGroup from '@/components/buttonsGroup/FilterButtonsGroup'
+import { selectAuthDetails } from '@/redux/features/authDetailsSlice'
+import { useFilter } from '@/hooks/useFilter'
 
 export const FilterBar = ({ updateViewModeSetting }: { updateViewModeSetting: (mode: View) => void }) => {
-  const [searchText, setSearchText] = useState('')
-
-  const { view } = useSelector(selectTaskBoard)
-
-  const { assignee } = useSelector(selectTaskBoard)
+  const [filterOptions, setFilterOptions] = useState<IFilterOptions>({
+    [FilterOptions.ASSIGNEE]: '',
+    [FilterOptions.KEYWORD]: '',
+    [FilterOptions.TYPE]: 'all',
+  })
+  const handleFilterOptionsChange = (optionType: FilterOptions, newValue: string | null) => {
+    setFilterOptions((prevOptions) => ({
+      ...prevOptions,
+      [optionType]: newValue,
+    }))
+  }
+  const [activeButtonIndex, setActiveButtonIndex] = useState<number>(3)
+  const { view, assignee } = useSelector(selectTaskBoard)
+  const { tokenPayload } = useSelector(selectAuthDetails)
 
   const { renderingItem: _assigneeValue, updateRenderingItem: updateAssigneeValue } = useHandleSelectorComponent({
     item: assignee[0],
     type: SelectorType.ASSIGNEE_SELECTOR,
   })
 
+  useFilter(filterOptions)
+
+  const filterButtons = [
+    {
+      name: 'My tasks',
+      onClick: async (index: number) => {
+        handleFilterOptionsChange(FilterOptions.TYPE, tokenPayload?.internalUserId as string)
```

Comment:
This is an unfortunate typecast, it's very clear `tokenPayload?.internalUserId` _can_ be undefined. But we're telling the compiler it's definitely a string with the `as string`. I recommend doing a runtime check for the `internalUserId` rather than casting the type.

Reply:
@arpandhakal You can use the IUToken schema in `types/common.ts` to parse this to make sure this token payload is one for an internal user